GeneralDQLIT integration test error

I found that the GeneralDQLIT test in the integration test has the following exceptions.


```Error: -13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:12.847 [ShardingSphere-Command-3] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 0
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:13.219 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:12.869 [ShardingSphere-Command-2] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 0
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:12.948 [ShardingSphere-Command-3] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 0
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:13.220 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:12.953 [ShardingSphere-Command-2] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 0
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : .java:98)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:13.221 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:13.222 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:13.222 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.010 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 4
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:14.224 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.018 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 4
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:51)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.043 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 4
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:50)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:14.225 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.047 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 4
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:50)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:14.226 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:14.228 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.067 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 5
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:50)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.LimitPaginationContextEngine.createPaginationContext(LimitPaginationContextEngine.java:38)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.s
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : elect.pagination.engine.PaginationContextEngine.createPaginationContext(PaginationContextEngine.java:50)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext.<init>(SelectStatementContext.java:95)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.getDMLStatementContext(SQLStatementContextFactory.java:125)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.SQLStatementContextFactory.newInstance(SQLStatementContextFactory.java:108)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.getProjectionCount(MySQLComStmtPrepareExecutor.java:77)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.binary.prepare.MySQLComStmtPrepareExecutor.execute(MySQLComStmtPrepareExecutor.java:68)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:98)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.lang.Thread.run(Thread.java:748)
Error: -13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : [ERROR] 2021-08-13 13:47:14.074 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : java.lang.IndexOutOfBoundsException: Index: 5
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at java.util.Collections$EmptyList.get(Collections.java:4454)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.getValue(PaginationContext.java:56)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at org.apache.shardingsphere.infra.binder.segment.select.pagination.PaginationContext.<init>(PaginationContext.java:50)
2021-08-13 13:47:14.229 [GeneralDQLIT-ShardingSphere-Proxy] : 	at ```

--- a/shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/pagination/PaginationContext.java
+++ b/shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/pagination/PaginationContext.java
@@ -18,6 +18,7 @@
 package org.apache.shardingsphere.infra.binder.segment.select.pagination;
 
 import lombok.Getter;
+import org.apache.commons.collections4.CollectionUtils;
 import org.apache.shardingsphere.infra.binder.statement.dml.SelectStatementContext;
 import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.pagination.NumberLiteralPaginationValueSegment;
 import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.pagination.PaginationValueSegment;
@@ -47,13 +48,13 @@ public final class PaginationContext {
         hasPagination = null != offsetSegment || null != rowCountSegment;
         this.offsetSegment = offsetSegment;
         this.rowCountSegment = rowCountSegment;
-        actualOffset = null == offsetSegment ? 0 : getValue(offsetSegment, parameters);
+        actualOffset = null == offsetSegment ? 0L : getValue(offsetSegment, parameters);
         actualRowCount = null == rowCountSegment ? null : getValue(rowCountSegment, parameters); 
     }
     
     private long getValue(final PaginationValueSegment paginationValueSegment, final List<Object> parameters) {
         if (paginationValueSegment instanceof ParameterMarkerPaginationValueSegment) {
-            Object obj = parameters.get(((ParameterMarkerPaginationValueSegment) paginationValueSegment).getParameterIndex());
+            Object obj = CollectionUtils.isEmpty(parameters) ? 0L : parameters.get(((ParameterMarkerPaginationValueSegment) paginationValueSegment).getParameterIndex());
             return obj instanceof Long ? (long) obj : (int) obj;
         } else {
             return ((NumberLiteralPaginationValueSegment) paginationValueSegment).getValue();

